File: leetcode/100-150/117-isPowerOfTwo.java
    public boolean isPowerOfTwo(int n) {
        int count = 0;
        while (n > 0) {
            if ((n & 1) == 1) count++;
            n >>= 1;
        }
        return count == 1;
    }

    public boolean isPowerOfTwo2(int n) {
        return n>0 && (n&(n-1))==0;
    }

    public boolean isPowerOfTwo3(int n) {
        return n>0 && Integer.bitCount()==1;
    }

File: leetcode/100-150/117-isPowerOfTwo.py
def isPowerOfTwo(self, n):
    return n >= 0 and bin(n).count('1') == 1

def isPowerOfTwo2(self, n):
    return n>0 and (n&(n-1))==0

def isPowerOfTwo3(self, n):
    while n!=0 and n%2==0: n/=2
    return n==1

def isPowerOfTwo4(self, n):
    return n>0 and (n==1 or (n%2==0 and isPowerOfTwo(self, n/2)))
print(isPowerOfTwo(None, 16))